### PR TITLE
Add demo form, add theme customization

### PR DESCRIPTION
**Pull request by RJFelix**
Draft
Normal
Closes DEV-NNN

- Add and export a demo form React component that is attached to a stubbed-out Userfront and does not interact with `window.Userfront`
  - [x] TODO: let the client pass in an auth flow for the demo form
  - [ ] Future: let the client receive and handle auth events emitted by the form
- Add CSS-based theming capabilities in new `dynamic.css` file + controls for it in the `UniversalForm` component
  - The dynamic theme is derived from client-provided colors: `light`, `dark`, optionally `accent`, optionally `light-background`, optionally `dark-background`
  - The dynamic theme handles both light and dark modes, and can be forced to one or the other or set to auto-switch based on the user's preference (`@media prefers-color-scheme`)
  - To control size and spacing, there's a `size` parameter that can be set to `default`, `compact`, `spaced`, `mini`, and `large` 
  - The theme can set the font
  - There's some additional optional additions to change the look & feel of the form, applied as CSS classes
    - `ounded` for corners to be more rounded than the default
    - `squared` to remove corner rounding altogether
    - `hideSecuredMessage` to hide the "secured by Userfront" branding
    - `gradientButtons` to add a subtle interactive gradient effect to buttons
    - `raisedButtons` to make the buttons pseudo-3D, like operating system buttons
      - [ ] Future: make `gradientButtons` and `raisedButtons` play nicely together (currently `raisedButtons` just overrides `gradientButtons`, but the gradient effect actually looks better on a raised background
    - `dottedOutlines` for focus/active outlines to be dotted lines (goes well with `raisedButtons`!)
  - While I put some effort into providing a sensible and functional menu of theming options, the capability is the point of this PR, and the specific options available are just a starting point - we can and should iterate on this
  - [x] TODO: apply theme to LogoutButton as appropriate
  - [ ] Todo in a follow up: Sass-ifiy the CSS, since it's now complex enough for Sass to pay real dividends

### CODE REVIEW

File: site/src/pages/DemoSignupForm.jsx [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        </label>
+        <input
+          name="useDarkBackgroundColor"
+          type="checkbox"
+          checked={!!darkBackgroundColor}
+          onChange={(evt) => {
+            if (evt.target.checked) {
+              setLightBackgroundColor("#ffffff");
+            } else {
+              setLightBackgroundColor(undefined);
```
Comment (drbrady8800) [inline site/src/pages/DemoSignupForm.jsx:121]:
```suggestion
              setDarkBackgroundColor("#ffffff");
            } else {
              setDarkBackgroundColor(undefined);
```

File: package/src/components/LogoutButton.jsx [hunk 2]
```diff
@@ -5,6 +5,19 @@ import { callUserfront } from "../services/userfront";
...
+ * @param {object=} props.theme.colors - theme colors
+ * @param {string=} props.theme.colors.accent - accent color to use when deriving color scheme (optional)
+ * @param {string=} props.theme.colors.lightBackground - background color for light mode (optional)
+ * @param {string=} props.theme.colors.darkBackground - background color for dark mode (optional)
+ * @param {string=} props.theme.fontFamily - CSS font family to use for the form
+ * @param {object=} props.theme.options - additional options to modify the form's appearance
+ * @param {boolean=} props.theme.options.rounded - make form elements appear more rounded generally
+ * @param {boolean=} props.theme.options.squared - make form elements appear more squared-off generally
+ * @param {boolean=} props.theme.options.gradientButtons - add an interactive gradient to buttons
+ * @param {boolean=} props.theme.options.hideSecuredMessage - hide the "secured by Userfront" message
```
Comment (drbrady8800) [inline package/src/components/LogoutButton.jsx:20]: dottedOutline is not documented here (or the other forms)

**Review by drbrady8800 — COMMENTED** [review]
Looks good! Aside from the small comments I had a question around default.css. It seems like it has been replaced everywhere except `package/.storybook/preview.cjs` and in the blurb around CSS in the README. Was wondering if this was intentional and if not if we should get rid of the default.css file?

**Comment by RJFelix** [comment]
> Looks good! Aside from the small comments I had a question around default.css. It seems like it has been replaced everywhere except `package/.storybook/preview.cjs` and in the blurb around CSS in the README. Was wondering if this was intentional and if not if we should get rid of the default.css file?

Good catches, thank you! I addressed your other comments as well.

**Review by drbrady8800 — APPROVED** [review]
LGTM!

**Pull request merged**